cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -318,7 +318,10 @@ public void performRunLevelStop(HttpRequest request, HttpResponder responder,
 
   private int validateAndGetGracefulShutdownSecsInt(String gracefulShutdownSecs) {
     try {
-      if (gracefulShutdownSecs == null || gracefulShutdownSecs.isEmpty()) {
+      if (gracefulShutdownSecs == null) {
+        return (int) TimeUnit.MINUTES.toSeconds(3);
```

Comment:
It's a little cleaner if there isn't any logic in the http handlers and the logic instead lives in the ProgramLifecycleService. Can we change the graceful time in the service to be a nullable int and just have the handler pass in a null if no graceful time is given.

It would also be good to add a property so that the time can be configured. Can you add a property to cdap-default.xml and then read the value from the cConf in the ProgramLifecycleService.